File: app/src/main/java/com/softwareoverflow/hiitultimate/database/dto/WorkoutSet.java
package com.softwareoverflow.hiitultimate.database.dto;

import androidx.room.Embedded;

import com.softwareoverflow.hiitultimate.database.entity.ExerciseTypeEntity;
import com.softwareoverflow.hiitultimate.database.entity.WorkoutSetEntity;

/**
 *  A WorkoutSetEntity consists of a 'work' period followed by a 'rest' period (AKA a 'rep').
 *  The 'rep' is then  repeated a set number of times before being followed by
 */
public class WorkoutSet{

    @Embedded
    private WorkoutSetEntity workoutSet;

    @Embedded
    private ExerciseTypeEntity exerciseTypeEntity;

    public WorkoutSetEntity getWorkoutSet() {
        return workoutSet;
    }

    public ExerciseTypeEntity getExerciseTypeEntity() {
        return exerciseTypeEntity;
    }

    public void setWorkoutSet(WorkoutSetEntity workoutSet) {
        this.workoutSet = workoutSet;
    }

    public void setExerciseTypeEntity(ExerciseTypeEntity exerciseTypeEntity) {
        this.exerciseTypeEntity = exerciseTypeEntity;
    }
}

File: app/src/main/java/com/softwareoverflow/hiit_trainer/data/WorkoutSet.kt
package com.softwareoverflow.hiit_trainer.data

import androidx.room.Embedded
import androidx.room.Relation
import com.softwareoverflow.hiit_trainer.data.entity.ExerciseTypeEntity
import com.softwareoverflow.hiit_trainer.data.entity.WorkoutSetEntity

data class WorkoutSet(
    @Embedded var workoutSet: WorkoutSetEntity,

    @Relation(
        parentColumn = "exerciseTypeId",
        entityColumn = "id"
    ) var exerciseType: ExerciseTypeEntity
)